crates/nu-parser/src/flatten.rs

```diff
@@ -624,6 +624,14 @@ pub fn flatten_pattern(match_pattern: &MatchPattern) -> Vec<(Span, FlatShape)> {
         Pattern::Value(_) => {
             output.push((match_pattern.span, FlatShape::MatchPattern));
         }
+        Pattern::Type(_, var_id) => match var_id {
+            Some(var_id) => {
+                output.push((match_pattern.span, FlatShape::VarDecl(*var_id)));
+            }
+            None => {
+                output.push((match_pattern.span, FlatShape::MatchPattern));
+            }
+        },
```

Comment:
Does it make sense to flatten this just into one flat shape or would we need to split that up into two for the left hand side and the type?

Reply:
I was not sure about that one. Maybe its better to do so yeah !